integration/update_compat_test.go

```diff
@@ -143,6 +151,254 @@ func testUpdateCompat(t *testing.T, testCases map[string]updateCompatTestCase) {
+							compatErr = compatCollection.Database().RunCommand(ctx, compatCommand).Decode(&compatUpdateRes)
+
+							if targetErr != nil {
+								t.Logf("Target error: %v", targetErr)
+								targetErr = UnsetRaw(t, targetErr)
+								compatErr = UnsetRaw(t, compatErr)
+
+								AssertMatchesCommandError(t, compatErr, targetErr)
+							} else {
+								require.NoError(t, compatErr, "compat error; target returned no error")
+							}
+
+							if targetUpdateRes != nil || compatUpdateRes != nil {
+								nonEmptyResults = true
+							}
+
+							assert.Equal(t, compatUpdateRes, targetUpdateRes)
+
+							if isMulti, ok := multi.(bool); ok && isMulti {
+								// if multi == false, an item updated by compat and target are different.
+
+								opts := options.Find().SetSort(bson.D{{"_id", 1}})
+								targetCursor, targetErr := targetCollection.Find(ctx, filter, opts)
+								compatCursor, compatErr := compatCollection.Find(ctx, filter, opts)
+
+								if targetCursor != nil {
+									defer targetCursor.Close(ctx)
+								}
+								if compatCursor != nil {
+									defer compatCursor.Close(ctx)
+								}
+
+								if targetErr != nil {
+									t.Logf("Target error: %v", targetErr)
+									AssertMatchesCommandError(t, compatErr, targetErr)
+
+									return
+								}
+								require.NoError(t, compatErr, "compat error; target returned no error")
+
+								var targetRes, compatRes []bson.D
+								require.NoError(t, targetCursor.All(ctx, &targetRes))
+								require.NoError(t, compatCursor.All(ctx, &compatRes))
+
+								t.Logf("Compat (expected) IDs: %v", CollectIDs(t, compatRes))
+								t.Logf("Target (actual)   IDs: %v", CollectIDs(t, targetRes))
+								AssertEqualDocumentsSlice(t, compatRes, targetRes)
+							}
+						})
+					}
+				})
+			}
+
+			switch tc.resultType {
+			case nonEmptyResult:
+				assert.True(t, nonEmptyResults, "expected non-empty results (some documents should be modified)")
+			case emptyResult:
+				assert.False(t, nonEmptyResults, "expected empty results (no documents should be modified)")
+			default:
+				t.Fatalf("unknown result type %v", tc.resultType)
+			}
+		})
+	}
+}
+
+// updateCurrentDateCompatTestCase describes update current date compatibility test case.
+type updateCurrentDateCompatTestCase struct {
+	skip       string                   // skips test if non-empty
+	paths      []types.Path             // paths to check after update
+	update     bson.D                   // required
+	filter     bson.D                   // defaults to bson.D{{"_id", id}}
+	resultType compatTestCaseResultType // defaults to nonEmptyResult
+}
+
+// testUpdateCompat tests update compatibility test cases for current date.
+// It checks current date in compat and target are within acceptable difference.
+func testUpdateCurrentDateCompat(t *testing.T, testCases map[string]updateCurrentDateCompatTestCase) {
```

Comment:
I can't find any `currentDate` test cases. Is this for the future?

Reply:
Please see the change in [integration/update_field_compat_test.go](https://github.com/FerretDB/FerretDB/pull/1659/files#diff-02beb83969c6feb8b48d15aebcd299647e0aa2e8b1e5c531e74eb2c4f636ad85)

There are a lot of changes in the file and Github does `Large diffs are not rendered by default.` :see_no_evil:

Reply:
Oh, right, thanks!